File: client-module-server/src/main/java/br/edu/les/module/client/domain/Usuario.java
package br.edu.les.module.client.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.stereotype.Component;

import javax.persistence.*;
import java.io.Serializable;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Component
@Entity
@Table(name = "usuario")
public class Usuario extends EntidadeDominio implements Serializable, IEntidade {
	
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "usu_id")
	private Integer id;

	@Column(name = "usu_codigo")
	private String codigo;

	@Column(name = "usu_email")
	private String email;

	@Column(name = "usu_senha")
	private String password;

	@Transient
	private String rePassword;

	@ManyToOne()
	@JoinColumn(name = "usu_status_id")
	private Status status;

	@ManyToOne()
	@JoinColumn(name = "usu_tipo_usuario_id")
	private TipoUsuario tipoUsuario;

	@OneToOne()
	@JoinColumn(name = "usu_cli_id")
	private Cliente cliente;
}

File: client-module-server-node/src/models/Usuario.ts
import { Column, Entity, JoinColumn, ManyToOne, PrimaryColumn, PrimaryGeneratedColumn } from "typeorm";
import { Status } from "./Status";
import { TipoUsuario } from "./TipoUsuario";

@Entity("usuario")
class Usuario {

    @PrimaryGeneratedColumn({name: "usu_id"})
    readonly id: number;

    @Column({name: "usu_email"})
    email: string;

    @Column({name: "usu_senha"})
	senha: string;

	reSenha: string;

	@ManyToOne(type => Status, {eager: true})
    @JoinColumn({name: "usu_status_id"})
	status: Status;

	@ManyToOne(type => TipoUsuario, tipoUsuario => tipoUsuario.usuarios, {eager: true})
    @JoinColumn({name: "usu_tipo_usuario_id"})
    tipoUsuario: TipoUsuario;
}

export { Usuario };